src/auth/src/credentials/user_credential.rs

```diff
@@ -512,8 +524,9 @@ mod test {
             refresh_token: "test-refresh-token".to_string(),
             endpoint: endpoint,
         };
+        let cached_token_provider = TokenCache::new(token_provider);
```

Comment:
optional: I am not sure this is adding anything to the test other than distraction. I might revert these changes here and elsewhere.

Reply:
The value it adds is that we validate whether all types of success and error responses get propagated correctly out of the token cache.

(But that is something ideally tested in the token cache itself, which we are doing. So I agree, let me revert these to directly token provider)